sycl/doc/extensions/proposed/sycl_ext_oneapi_device_architecture.asciidoc

```diff
@@ -0,0 +1,638 @@
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+[comment]
+--
+_Add the following paragraph when this specification becomes "experimental"._
+
+There are important limitations with the DPC++ implementation of this
+experimental extension.  In particular, this extension may only be used when
+the application is compiled in AOT mode.  See the section below titled
+"Limitations with the experimental version" for a full description of the
+limitations.
+--
+
+
+== Overview
+
+This extension provides a way for device code to query the device architecture
+on which it is running.  This is similar to the
+link:./sycl_ext_oneapi_device_if.asciidoc[sycl_ext_oneapi_device_if] extension
+except the comparison is for the device's architecture not the device's
+aspects.  In some cases, low-level application code can use special features or
+do specific optimizations depending on the device architecture, and this
+extension enables such applications.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_DEVICE_ARCHITECTURE` to one of the values defined in the
+table below.  Applications can test for the existence of this macro to
+determine if the implementation supports this feature, or applications can test
+the macro's value to determine which of the extension's features the
+implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+
+|2
+|NVIDIA and AMD architectures are added.
+|===
+
+=== New enumeration of architectures
+
+This extension adds a new enumeration of the architectures that can be tested.
+
+```
+namespace sycl::ext::oneapi::experimental {
+
+enum class architecture : /* unspecified */ {
+  x86_64,
+  intel_gpu_bdw,
+  intel_gpu_skl,
+  intel_gpu_kbl,
+  intel_gpu_cfl,
+  intel_gpu_apl,
+  intel_gpu_glk,
+  intel_gpu_whl,
+  intel_gpu_aml,
+  intel_gpu_cml,
+  intel_gpu_icllp,
+  intel_gpu_ehl,
+  intel_gpu_tgllp,
+  intel_gpu_rkl,
+  intel_gpu_adl_s,
+  intel_gpu_rpl_s,
+  intel_gpu_adl_p,
+  intel_gpu_adl_n,
+  intel_gpu_dg1,
+  intel_gpu_acm_g10,
+  intel_gpu_acm_g11,
+  intel_gpu_acm_g12,
+  intel_gpu_pvc,
+
+  intel_gpu_8_0_0 = intel_gpu_bdw,
+  intel_gpu_9_0_9 = intel_gpu_skl,
+  intel_gpu_9_1_9 = intel_gpu_kbl
+  intel_gpu_9_2_9 = intel_gpu_cfl,
+  intel_gpu_9_3_0 = intel_gpu_apl,
+  intel_gpu_9_4_0 = intel_gpu_glk,
+  intel_gpu_9_5_0 = intel_gpu_whl,
+  intel_gpu_9_6_0 = intel_gpu_aml,
+  intel_gpu_9_7_0 = intel_gpu_cml,
+  intel_gpu_11_0_0 = intel_gpu_icllp,
+  intel_gpu_11_2_0 = intel_gpu_ehl,
+  intel_gpu_12_0_0 = intel_gpu_tgllp,
+  intel_gpu_12_10_0 = intel_gpu_dg1,
+
+  nvidia_gpu_sm20,
```

Comment:
All these "nvidia" and "amd" enumerators should go before the "alias" enumerators like `intel_gpu_8_0_0`.  Otherwise, the "nvidia" and "amd" enumerators will alias some of the "intel" ones.